### PR TITLE
P1622R3 Mandating the Standard Library: Clause 32 - Thread support library

### LINKED ISSUE #3433: [2019-11 LWG Motion 31]  P1622R3 Mandating the Standard Library: Clause 32 - Thread support library

### PR DESCRIPTION
**Pull request by zygoloid**
Fixes #3433.
Fixes cplusplus/papers#394.

### CODE REVIEW

File: source/threads.tex [hunk 1]
```diff
@@ -7071,14 +7028,19 @@
...
-       @\placeholdernc{decay-copy}@(std::forward<Args>(args))...)     // see \ref{func.require}, \ref{thread.thread.constr}
-\end{codeblock}
-shall be a valid expression.
+\mandates
+The following are all \tcode{true}:
+\begin{itemize}
+\item \tcode{is_constructible<decay_t<F>, F>},
+\item \tcode{(is_constructible<decay_t<Args>, Args> \&\&...)},
+\item \tcode{is_move_constructible<decay_t<F>>},
+\item \tcode{(is_move_constructible<decay_t<Args>> \&\&...)}, and
```
Comment (johelegp) [inline source/threads.tex]:
```suggestion
\item \tcode{(is_move_constructible_v<decay_t<Args>> \&\&...)}, and
```

File: source/threads.tex [hunk 2]
```diff
@@ -7071,14 +7028,19 @@
...
-@\placeholdernc{INVOKE}@(@\placeholdernc{decay-copy}@(std::forward<F>(f)),
-       @\placeholdernc{decay-copy}@(std::forward<Args>(args))...)     // see \ref{func.require}, \ref{thread.thread.constr}
-\end{codeblock}
-shall be a valid expression.
+\mandates
+The following are all \tcode{true}:
+\begin{itemize}
+\item \tcode{is_constructible<decay_t<F>, F>},
+\item \tcode{(is_constructible<decay_t<Args>, Args> \&\&...)},
+\item \tcode{is_move_constructible<decay_t<F>>},
```
Comment (johelegp) [inline source/threads.tex]:
```suggestion
\item \tcode{is_move_constructible_v<decay_t<F>>},
```

File: source/threads.tex [hunk 3]
```diff
@@ -7071,14 +7028,19 @@
...
-\begin{codeblock}
-@\placeholdernc{INVOKE}@(@\placeholdernc{decay-copy}@(std::forward<F>(f)),
-       @\placeholdernc{decay-copy}@(std::forward<Args>(args))...)     // see \ref{func.require}, \ref{thread.thread.constr}
-\end{codeblock}
-shall be a valid expression.
+\mandates
+The following are all \tcode{true}:
+\begin{itemize}
+\item \tcode{is_constructible<decay_t<F>, F>},
+\item \tcode{(is_constructible<decay_t<Args>, Args> \&\&...)},
```
Comment (johelegp) [inline source/threads.tex]:
```suggestion
\item \tcode{(is_constructible_v<decay_t<Args>, Args> \&\&...)},
```

File: source/threads.tex [hunk 4]
```diff
@@ -7071,14 +7028,19 @@
...
-\oldconcept{MoveConstructible} requirements, and
-\begin{codeblock}
-@\placeholdernc{INVOKE}@(@\placeholdernc{decay-copy}@(std::forward<F>(f)),
-       @\placeholdernc{decay-copy}@(std::forward<Args>(args))...)     // see \ref{func.require}, \ref{thread.thread.constr}
-\end{codeblock}
-shall be a valid expression.
+\mandates
+The following are all \tcode{true}:
+\begin{itemize}
+\item \tcode{is_constructible<decay_t<F>, F>},
```
Comment (johelegp) [inline source/threads.tex]:
```suggestion
\item \tcode{is_constructible_v<decay_t<F>, F>},
```

File: source/threads.tex [hunk 5]
```diff
@@ -119,13 +119,13 @@
 time $C_n \geq C_t$, i.e., waking at $C_t + D_i + D_m$.
```
Comment (zygoloid) [inline source/threads.tex]: The note 7 lines above is no longer a note in the wording paper, but there is no "removed" markup for this change. I've checked with the paper author, and this note was indeed intended to be made normative by this paper (as requested by LWG).

**Pull request merged**